### ISSUE TITLE
SNZB-02 fails to send temperature msTemperatureMeasurement failed (AREQ - ZDO - bindRsp after 10000ms)

### ISSUE THREAD

**Issue by tvrulesme** [Koenkk/zigbee2mqtt#8971]
## Debug info
Zigbee2MQTT version: version 1.21.2 (commit #44922120)
Adapter hardware: CC2531
Adapter firmware version: CC2531_DEFAULT_20201127

Supported device SNZB-02 pairs successfully but shows the following error when it attempts to broadcast a temperature reading
`Zigbee2MQTT:error 2021-10-03 10:23:10: Configure failed: Error: Bind 0x00124b0022cf704d/1 msTemperatureMeasurement from '0x00124b00193691c1/1' failed (AREQ - ZDO - bindRsp after 10000ms)`

configuration.yaml:
```
homeassistant: false
permit_join: true
mqtt:
  base_topic: zigbee2mqtt
  server: mqtt://192.168.0.2
  user: mqtt_user
  password: Penguin1
serial:
  port: /dev/ttyACM0
devices:
  '0x00124b0022cf704d':
    friendly_name: '01-humidity'
    retain: true
```

logs:
```
Zigbee2MQTT:info  2021-10-03 10:12:49: Starting Zigbee2MQTT version 1.21.2 (commit #44922120)
Zigbee2MQTT:info  2021-10-03 10:12:49: Starting zigbee-herdsman (0.13.149)
Zigbee2MQTT:info  2021-10-03 10:12:58: zigbee-herdsman started (restored)
Zigbee2MQTT:info  2021-10-03 10:12:58: Coordinator firmware version: '{"meta":{"maintrel":3,"majorrel":2,"minorrel":6,"product":0,"revision":20201127,"transportrev":2},"type":"zStack12"}'
Zigbee2MQTT:info  2021-10-03 10:12:58: Currently 0 devices are joined:
.......

info  2021-10-03 10:18:39: Device '0x00124b0022cf704d' joined
info  2021-10-03 10:18:39: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x00124b0022cf704d","ieee_address":"0x00124b0022cf704d"},"type":"device_joined"}'
info  2021-10-03 10:18:39: Starting interview of '0x00124b0022cf704d'
info  2021-10-03 10:18:39: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x00124b0022cf704d","ieee_address":"0x00124b0022cf704d","status":"started"},"type":"device_interview"}'
info  2021-10-03 10:18:39: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":{"friendly_name":"0x00124b0022cf704d"},"type":"device_connected"}'
info  2021-10-03 10:18:39: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"interview_started","meta":{"friendly_name":"0x00124b0022cf704d"},"type":"pairing"}'
info  2021-10-03 10:18:40: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"0x00124b0022cf704d","ieee_address":"0x00124b0022cf704d"},"type":"device_announce"}'
info  2021-10-03 10:18:40: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"announce","meta":{"friendly_name":"0x00124b0022cf704d"},"type":"device_announced"}'
info  2021-10-03 10:18:40: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":[],"type":"groups"}'
info  2021-10-03 10:18:40: MQTT publish: topic 'zigbee2mqtt/bridge/config/devices', payload '[{"dateCode":"20201127","friendly_name":"Coordinator","ieeeAddr":"0x00124b00193691c1","lastSeen":1633252720100,"networkAddress":0,"softwareBuildID":"zStack12","type":"Coordinator"},{"description":"-","friendly_name":"0x00124b0022cf704d","ieeeAddr":"0x00124b0022cf704d","lastSeen":1633252720050,"networkAddress":59037,"type":"Unknown","vendor":"-"}]'
info  2021-10-03 10:18:40: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":[{"dateCode":"20201127","friendly_name":"Coordinator","ieeeAddr":"0x00124b00193691c1","lastSeen":1633252720166,"networkAddress":0,"softwareBuildID":"zStack12","type":"Coordinator"},{"description":"-","friendly_name":"0x00124b0022cf704d","ieeeAddr":"0x00124b0022cf704d","lastSeen":1633252720050,"manufacturerID":0,"networkAddress":59037,"type":"EndDevice","vendor":"-"}],"type":"devices"}'
info  2021-10-03 10:18:45: Successfully interviewed '0x00124b0022cf704d', device has successfully been paired
info  2021-10-03 10:18:45: Device '0x00124b0022cf704d' is supported, identified as: SONOFF Temperature and humidity sensor (SNZB-02)
info  2021-10-03 10:18:45: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"definition":{"description":"Temperature and humidity sensor","exposes":[{"access":1,"description":"Remaining battery in %","name":"battery","property":"battery","type":"numeric","unit":"%","value_max":100,"value_min":0},{"access":1,"description":"Measured temperature value","name":"temperature","property":"temperature","type":"numeric","unit":"°C"},{"access":1,"description":"Measured relative humidity","name":"humidity","property":"humidity","type":"numeric","unit":"%"},{"access":1,"description":"Voltage of the battery in millivolts","name":"voltage","property":"voltage","type":"numeric","unit":"mV"},{"access":1,"description":"Link quality (signal strength)","name":"linkquality","property":"linkquality","type":"numeric","unit":"lqi","value_max":255,"value_min":0}],"model":"SNZB-02","supports_ota":false,"vendor":"SONOFF"},"friendly_name":"0x00124b0022cf704d","ieee_address":"0x00124b0022cf704d","status":"successful","supported":true},"type":"device_interview"}'
info  2021-10-03 10:18:45: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"interview_successful","meta":{"description":"Temperature and humidity sensor","friendly_name":"0x00124b0022cf704d","model":"SNZB-02","supported":true,"vendor":"SONOFF"},"type":"pairing"}'
info  2021-10-03 10:18:45: Changed device specific options of '0x00124b0022cf704d' ({"retain":true})

......

Zigbee2MQTT:info  2021-10-03 10:23:00: Device '01-humidity' joined
Zigbee2MQTT:info  2021-10-03 10:23:00: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"01-humidity","ieee_address":"0x00124b0022cf704d"},"type":"device_joined"}'
Zigbee2MQTT:info  2021-10-03 10:23:00: Configuring '01-humidity'
Zigbee2MQTT:info  2021-10-03 10:23:00: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":{"friendly_name":"01-humidity"},"type":"device_connected"}'
Zigbee2MQTT:info  2021-10-03 10:23:00: MQTT publish: topic 'zigbee2mqtt/bridge/event', payload '{"data":{"friendly_name":"01-humidity","ieee_address":"0x00124b0022cf704d"},"type":"device_announce"}'
Zigbee2MQTT:info  2021-10-03 10:23:00: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":"announce","meta":{"friendly_name":"01-humidity"},"type":"device_announced"}'
Zigbee2MQTT:info  2021-10-03 10:23:00: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":[],"type":"groups"}'
Zigbee2MQTT:info  2021-10-03 10:23:00: MQTT publish: topic 'zigbee2mqtt/bridge/config/devices', payload '[{"dateCode":"20201127","friendly_name":"Coordinator","ieeeAddr":"0x00124b00193691c1","lastSeen":1633252980641,"networkAddress":0,"softwareBuildID":"zStack12","type":"Coordinator"},{"dateCode":"20201026","description":"Temperature and humidity sensor","friendly_name":"01-humidity","hardwareVersion":1,"ieeeAddr":"0x00124b0022cf704d","lastSeen":1633252980584,"manufacturerID":0,"manufacturerName":"eWeLink","model":"SNZB-02","modelID":"TH01","networkAddress":35923,"powerSource":"Battery","type":"EndDevice","vendor":"SONOFF"}]'
Zigbee2MQTT:info  2021-10-03 10:23:00: MQTT publish: topic 'zigbee2mqtt/bridge/log', payload '{"message":[{"dateCode":"20201127","friendly_name":"Coordinator","ieeeAddr":"0x00124b00193691c1","lastSeen":1633252980659,"networkAddress":0,"softwareBuildID":"zStack12","type":"Coordinator"},{"dateCode":"20201026","description":"Temperature and humidity sensor","friendly_name":"01-humidity","hardwareVersion":1,"ieeeAddr":"0x00124b0022cf704d","lastSeen":1633252980584,"manufacturerID":0,"manufacturerName":"eWeLink","model":"SNZB-02","modelID":"TH01","networkAddress":35923,"powerSource":"Battery","type":"EndDevice","vendor":"SONOFF"}],"type":"devices"}'
Zigbee2MQTT:error 2021-10-03 10:23:10: Configure failed: Error: Bind 0x00124b0022cf704d/1 msTemperatureMeasurement from '0x00124b00193691c1/1' failed (AREQ - ZDO - bindRsp after 10000ms)
Zigbee2MQTT:info  2021-10-03 10:23:10: Successfully configured '01-humidity'
Zigbee2MQTT:info  2021-10-03 10:27:01: MQTT publish: topic 'zigbee2mqtt/bridge/state', payload 'offline'
```

**Comment by qukla1000** [comment]
I have the same problem: https://github.com/Koenkk/zigbee2mqtt/issues/8979

**Comment by yang360** [comment]
Same problem here as well.

**Comment by sjorge** [comment]
Make sure to force remove the failed pairing before trying again, I have the best success with removing the battery and putting it back in, instead of holding the button.

**Comment by tvrulesme** [comment]
> Make sure to force remove the failed pairing before trying again, I have the best success with removing the battery and putting it back in, instead of holding the button.

Yes I have done this multiple times as well as deleting my database file and state files. Also tried with the dev branch with the same result

**Comment by qukla1000** [comment]
Me too. I tired add three new sensors and the result was the same all the time. A tried add multiple times also with force removing. I also deleted datatbase and reinstall zigbe2mqtt. I tired reflash CC26X2R1 but it didn't help. Also tried change USB cable to another.
I also added some tuya motion sensor - that works without any problem. The problem exists only with Sonoff SNZB-02.

**Comment by sjorge** [comment]
Are you pairing through a router or at the coordinator? I have a zzhp-lite prototype and all my pairings of my SNZB-02's were done through routers.

**Comment by tvrulesme** [comment]
> Are you pairing through a router or at the coordinator? I have a zzhp-lite prototype and all my pairings of my SNZB-02's were done through routers.

Assuming through coordinator. Is zzhp-lite hardware?

**Comment by tvrulesme** [comment]
> Are you pairing through a router or at the coordinator? I have a zzhp-lite prototype and all my pairings of my SNZB-02's were done through routers.

So my hardware is listed above it's pretty much the same as this: https://uk.rs-online.com/web/p/communication-wireless-development-tools/1977102/?cm_mmc=UK-PLA-DS3A-_-google-_-CSS_UK_EN_Raspberry_Pi_%26_Arduino_%26_Development_Tools_Whoop-_-Communication+%26+Wireless+Development+Tools_Whoop-_-1977102&matchtype=&pla-306759658968&gclid=CjwKCAjwzOqKBhAWEiwArQGwaEw5EcmHj-D38MUQm1z3uc8DZWEatV99cjjqob-V-9l6ZEpbq44ZjhoC4YAQAvD_BwE&gclsrc=aw.ds

**Comment by sjorge** [comment]
> Assuming through coordinator. Is zzhp-lite hardware?

Yeah zzh-lite is a CC2652P based coordinator, can you try pairing through a router instead? Most of my bulbs are Trådfri bulbs and a few Hue/Innr so I don't think the manufacturer should matter.

**Comment by tvrulesme** [comment]
> > Assuming through coordinator. Is zzhp-lite hardware?
> 
> Yeah zzh-lite is a CC2652P based coordinator, can you try pairing through a router instead? Most of my bulbs are Trådfri bulbs and a few Hue/Innr so I don't think the manufacturer should matter.

OK now I am confused. So you mean take the SNZB-02 away from the CC2531 and try and use my wifi router?

**Comment by qukla1000** [comment]
> Are you pairing through a router or at the coordinator? I have a zzhp-lite prototype and all my pairings of my SNZB-02's were done through routers.

I have only coordinator in my network.

**Comment by qukla1000** [comment]
> > > Assuming through coordinator. Is zzhp-lite hardware?
> > 
> > 
> > Yeah zzh-lite is a CC2652P based coordinator, can you try pairing through a router instead? Most of my bulbs are Trådfri bulbs and a few Hue/Innr so I don't think the manufacturer should matter.
> 
> OK now I am confused. So you mean take the SNZB-02 away from the CC2531 and try and use my wifi router?

Not wifi router. Zigbee router - if you have.

**Comment by tvrulesme** [comment]
> > > > Assuming through coordinator. Is zzhp-lite hardware?
> > > 
> > > 
> > > Yeah zzh-lite is a CC2652P based coordinator, can you try pairing through a router instead? Most of my bulbs are Trådfri bulbs and a few Hue/Innr so I don't think the manufacturer should matter.
> > 
> > 
> > OK now I am confused. So you mean take the SNZB-02 away from the CC2531 and try and use my wifi router?
> 
> Not wifi router. Zigbee router - if you have.

Ah no, I don't sorry

**Comment by sjorge** [comment]
> OK now I am confused. So you mean take the SNZB-02 away from the CC2531 and try and use my wifi router?

> Not wifi router. Zigbee router - if you have.

Yes, zigbee router... usually something that always has power like a bulb.

**Comment by sjorge** [comment]
> I have only coordinator in my network.

> Ah no, I don't sorry

Interesting, both of you don't have any routers in the network, I wonder if somehow due to this message don't get buffered as there are no routers to hold them.

@Koenkk unless I am mistaken that is how it usually works right, also why the IKEA blinds need the repeater or another router for the remote to be able to talk to the blinds... I wonder if that is an issue here too.

**Comment by qukla1000** [comment]
> > I have only coordinator in my network.
> 
> > Ah no, I don't sorry
> 
> Interesting, both of you don't have any routers in the network, I wonder if somehow due to this message don't get buffered as there are no routers to hold them.
> 
> @Koenkk unless I am mistaken that is how it usually works right, also why the IKEA blinds need the repeater or another router for the remote to be able to talk to the blinds... I wonder if that is an issue here too.

I also have other installation in other house. But zigbee2mqtt version is 1.18.1-1. There is also one coordinator, other sensors and few SNZB-02. It works without any problem. Joining and reporting states form sensors works seamlessly. It also works with CC26X2R1 bout i don't know with version of software.

**Comment by tvrulesme** [comment]
> > > I have only coordinator in my network.
> > 
> > 
> > > Ah no, I don't sorry
> > 
> > 
> > Interesting, both of you don't have any routers in the network, I wonder if somehow due to this message don't get buffered as there are no routers to hold them.
> > @Koenkk unless I am mistaken that is how it usually works right, also why the IKEA blinds need the repeater or another router for the remote to be able to talk to the blinds... I wonder if that is an issue here too.
> 
> I also have other installation in other house. But zigbee2mqtt version is 1.18.1-1. There is also one coordinator, other sensors and few SNZB-02. It works without any problem. Joining and reporting states form sensors works seamlessly. It also works with CC26X2R1 bout i don't know with version of software.

Went through some old hardware boxes and found a switch which is acting as a router. So cleaned out database, added switch, added SNZB-02 and no error messages **BUT** I do not see any data in the logs either. How often does yours generally publish?

**Comment by tvrulesme** [comment]
Right, battery in and out and I am seeing data :)

`Zigbee2MQTT:info  2021-10-04 13:42:11: MQTT published: topic 'zigbee2mqtt/humidity', payload '{"linkquality":89,"voltage":2900}'
Zigbee2MQTT:info  2021-10-04 13:42:12: MQTT published: topic 'zigbee2mqtt/humidity', payload '{"battery":85,"linkquality":89,"voltage":2900}'
Zigbee2MQTT:info  2021-10-04 13:42:12: Successfully configured 'humidity'
Zigbee2MQTT:info  2021-10-04 13:42:17: MQTT published: topic 'zigbee2mqtt/humidity', payload '{"battery":85,"linkquality":89,"temperature":23.74,"voltage":2900}'
Zigbee2MQTT:info  2021-10-04 13:42:17: MQTT published: topic 'zigbee2mqtt/humidity', payload '{"battery":85,"humidity":81.51,"linkquality":89,"temperature":23.74,"voltage":2900}'
Zigbee2MQTT:info  2021-10-04 13:42:27: MQTT published: topic 'zigbee2mqtt/humidity', payload '{"battery":85,"humidity":75.94,"linkquality":89,"temperature":23.74,"voltage":2900}'
Zigbee2MQTT:info  2021-10-04 13:42:37: MQTT published: topic 'zigbee2mqtt/humidity', payload '{"battery":85,"humidity":68.46,"linkquality":102,"temperature":23.74,"voltage":2900}'
Zigbee2MQTT:info  2021-10-04 13:42:47: MQTT published: topic 'zigbee2mqtt/humidity', payload '{"battery":85,"humidity":64.67,"linkquality":92,"temperature":23.74,"voltage":2900}'
Zigbee2MQTT:info  2021-10-04 13:42:58: MQTT published: topic 'zigbee2mqtt/humidity', payload '{"battery":85,"humidity":63.18,"linkquality":99,"temperature":23.74,"voltage":2900}'`

**Comment by qukla1000** [comment]
> > > > I have only coordinator in my network.
> > > 
> > > 
> > > > Ah no, I don't sorry
> > > 
> > > 
> > > Interesting, both of you don't have any routers in the network, I wonder if somehow due to this message don't get buffered as there are no routers to hold them.
> > > @Koenkk unless I am mistaken that is how it usually works right, also why the IKEA blinds need the repeater or another router for the remote to be able to talk to the blinds... I wonder if that is an issue here too.
> > 
> > 
> > I also have other installation in other house. But zigbee2mqtt version is 1.18.1-1. There is also one coordinator, other sensors and few SNZB-02. It works without any problem. Joining and reporting states form sensors works seamlessly. It also works with CC26X2R1 bout i don't know with version of software.
> 
> Went through some old hardware boxes and found a switch which is acting as a router. So cleaned out database, added switch, added SNZB-02 and no error messages **BUT** I do not see any data in the logs either. How often does yours generally publish?

So it looks similar. I can add sonoff, it shows in zigbe2mqtt but sensor don't publish any status even after pushing button.
Of course while i adding this sensor i have error like:
`sTemperatureMeasurement failed (AREQ - ZDO - bindRsp after 10000ms)`
Not always but most of the time.

**Comment by sjorge** [comment]
@tvrulesme certainly interesting, so a router in the mesh does seem to help. 
Mine don't immediately send data either, it goes something like this
1. force remove device
2. remove battery
3. enable join via a bulb in the room with the sensor
4. insert battery 
5. (check if the red led blinked, if not, hold the button until it does)
6. wait for z2m to pair successfully
7. (timeout error, back to 1.)

Then I think around ~15 min later it starts sending data, I think that might be some build in default when it first wakes up. Then it seems to follow whatever is configured in the reporting. Removing the battery and inserting it again once it is paired does seem to immediately have it publish battery and voltage, and the rest follow shortly after.


> So it looks similar. I can add sonoff, it shows in zigbe2mqtt but sensor don't publish any status even after pushing button. Of course while i adding this sensor i have error like: `sTemperatureMeasurement failed (AREQ - ZDO - bindRsp after 10000ms)` Not always but most of the time.

I'm guessing it will never send temperature if msTemperatureMeasurement reporting setup gave a timeout. I think that's also the first one it tries, so it probably never got around to setting up the others either, not having any data would make sense then.

I never had luck manually configuring the reporting on these devices, but you could also try that.
<img width="1426" alt="image" src="https://user-images.githubusercontent.com/379665/135857763-0b1cff4f-d238-40bb-963a-d926125e5ddf.png">


This is what the default reporting looks like.

**Comment by Koenkk** [comment]
@tvrulesme if I understand correctly, the device works fine (temperature etc is updated), the only error here is the configure failing?

@sjorge if that is the case we can add a try/catch to the configure (to suppress this error) as it seems not all devices need this.

**Comment by qukla1000** [comment]
> @tvrulesme if I understand correctly, the device works fine (temperature etc is updated), the only error here is the configure failing?
> 
> @sjorge if that is the case we can add a try/catch to the configure (to suppress this error) as it seems not all devices need this.

@Koenkk For me device don't work but shows in zigbe2mqtt. Devie don't raport there state. Error shows most f the time but not always. Pushing button on sensor don't update status. Device also isn't reporting its state after while.

**Comment by tvrulesme** [comment]
> @tvrulesme if I understand correctly, the device works fine (temperature etc is updated), the only error here is the configure failing?
> 

Yes this is correct @Koenkk, works with a router but not stand alone

**Comment by Koenkk** [comment]
@tvrulesme any chance you could provide a sniff when pairing via both router (working situation) and non-working situation (coordinator). https://www.zigbee2mqtt.io/how_tos/how_to_sniff_zigbee_traffic.html

**Comment by tvrulesme** [comment]
> @tvrulesme any chance you could provide a sniff when pairing via both router (working situation) and non-working situation (coordinator). https://www.zigbee2mqtt.io/how_tos/how_to_sniff_zigbee_traffic.html

Yeah absolutely. Will get on it tomorrow

**Comment by takthetank** [comment]
I have the same problem with SNZB-02 sensors, while other devices work fine.

**Comment by tvrulesme** [comment]
> @tvrulesme any chance you could provide a sniff when pairing via both router (working situation) and non-working situation (coordinator). https://www.zigbee2mqtt.io/how_tos/how_to_sniff_zigbee_traffic.html

Really sorry @Koenkk I got stuck at this point
```
pi@raspberrypi:~/flash_cc2531 $ /snap/bin/cc-tool.run -e -w sniffer_fw_cc2531.hex 
  CC Debugger device not found
```
This is on a raspberry pi.

**Comment by tvrulesme** [comment]
Brief update. Sadly the resolution above does not appear to be consistent. Tried a complete clean down removing all devices, adding the switch first followed by the SNZB-02 and again I see this error:

`Zigbee2MQTT:error 2021-10-05 11:25:59: Configure failed: Error: Bind 0x00124b0022cf704d/1 msTemperatureMeasurement from '0x00124b00193691c1/242' failed (AREQ - ZDO - bindRsp after 10000ms)
Zigbee2MQTT:info  2021-10-05 11:25:59: Successfully configured '0x00124b0022cf704d'` 

Please shout if there is anything useful I can send through

**Comment by sjorge** [comment]
You only enabled joining through the switch right?

**Comment by tvrulesme** [comment]
> You only enabled joining through the switch right?

Sorry @sjorge how would I do that?

**Comment by sjorge** [comment]
Easiest is using the frontend and hitting the `V` icon next to `Permit join (All)` then selecting the switch:
<img width="442" alt="image" src="https://user-images.githubusercontent.com/379665/136009319-2ddb357c-6859-4ce9-94bc-af2d87d3c176.png">

**Comment by tvrulesme** [comment]
> That should prevent the sensor from trying to use the coordinator.
![Screenshot 2021-10-05 at 11 54 56](https://user-images.githubusercontent.com/16368536/136010314-e9e8c166-9fdf-4500-9b2c-b5ccb1e760dc.png)
Only seeing Disable join. Is that right?

**Comment by tvrulesme** [comment]
![Screenshot 2021-10-05 at 12 02 11](https://user-images.githubusercontent.com/16368536/136011105-175bf5a4-d1b9-449a-934e-8600d7a42388.png)
Lined to coordinator. So adding it to router before must have just been luck

**Comment by sjorge** [comment]
Might have been because joining was still enabled from before you selected the switch?

**Comment by tvrulesme** [comment]
@sjorge have tried everything I can think of and I can't get it to join the switch instead of composer.

**Comment by tvrulesme** [comment]
Ok, finally got it to join the router (by physically moving both a long way)


<img width="433" alt="Screenshot 2021-10-05 at 13 35 01" src="https://user-images.githubusercontent.com/16368536/136023458-56910041-291a-4e5e-bc5d-2907f335efdf.png">
But the error message persists

```
Zigbee2MQTT:error 2021-10-05 13:31:59: Configure failed: Error: Bind 0x00124b0022cf704d/1 msTemperatureMeasurement from '0x00124b00193691c1/1' failed (AREQ - ZDO - bindRsp after 10000ms)
Zigbee2MQTT:info  2021-10-05 13:31:59: Successfully configured '0x00124b0022cf704d'
Zigbee2MQTT:info  2021-10-05 13:32:07: Starting network scan (includeRoutes 'false')
Zigbee2MQTT:info  2021-10-05 13:32:09: Network scan finished
```

**Comment by sjorge** [comment]
😞 so the earlier successful attempt was just random then

**Comment by tvrulesme** [comment]
> 😞 so the earlier successful attempt was just random then

Unfortunately yes. So sorry to raise hopes

**Comment by Koenkk** [comment]
I've ordered the device to further debug this issue.

**Comment by takthetank** [comment]
Hi @Koenkk I managed to sniff through windows, please find attached the z2m and sniffing logs.
[SNZB-02_logs.zip](https://github.com/Koenkk/zigbee2mqtt/files/7299485/SNZB-02_logs.zip)

Please tell me if you need something more!

**Comment by phobosdigg** [comment]
The same problem with this sensor.
The problem appeared after I changed the pan_id to 6755 in the zigbee2mqtt settings due to other problems.
Up to this point, the sensors worked fine.
What pan_id do you use in the settings zigbee2mqtt?

**Comment by takthetank** [comment]
I haven't set or changed the pan_id!

**Comment by grfernandes54** [comment]
I Tried to add 4 SNZB-02 and have the same problem, all sensors show N/A at the exposes

![image](https://user-images.githubusercontent.com/70862164/136640150-7ffd35c8-6995-406e-a079-8e98864b5d24.png)

**Comment by nickmcgill** [comment]
I have just downloaded the latest-dev docker image, and I can now pair the SNZB-02, no errors appear to be logged, but, to get it to report correctly I need to reconfigure it.
The reconfiguration must be done before renaming the device.
I hope this is of some help.

**Comment by deatondg** [comment]
@maxbause What on Earth? I've been having this same issue for days. I recently bought 2 new sensors, and they wouldn't work although my existing ones would re-pair just fine. This method totally worked for me. Very easy, and very strange.

**Comment by maxbause** [comment]
I'm happy that the method works for you too @degreal and @deatondg! :)

@sjorge I also think that this problem has something to do with the state of the sensor, but I haven't had the time yet to further investigate.

**Comment by nicolaferraro** [comment]
> Hey guys, I had the same problem: The sensor was connected, but I couldn't get any date out of it.
> 
> I've found a sloppy fix, but it worked for me:
> 
> 1. Make sure that the sensor is still connected to the bridge
> 2. Now open the back of the sensor and remove the button cell battery
> 3. In your zigbee2mqtt web interface, force remove the sensor
> 4. Take a piece of metal (I used a screwdriver tip) and short out the two battery contacts of the sensor (see attached file)
> 5. Now put the battery back in and wait for the sensor to connect
>    5.1 Note: Don't press the button on the sensor, let it connect by itself
> 
> After that, all of my 5 sensor worked perfectly. ![IMG_8794](https://user-images.githubusercontent.com/21057112/136665926-fed0fcc3-cc32-4146-b9e3-cf263f8b890f.JPG)

Hi @maxbause, I tried it but the sensor does not seem to connect back in step 5 (unless I hold the button again for a few secs). Should I just plug the battery and leave it alone for some minutes or reconnection happens immediately?

Does the light flash when you plug the battery again?

**Comment by garykrige** [comment]
I am also not managing to connect back (I'm using the latest dev docker image). I've left the device to connect for 20 minutes, but still no signs of connection without using the button - at which point it fails.

**Comment by valdeira888** [comment]
> Same problem here with my new SNZB-02 (I'm trying to add two of them but same error for both), here's the log line:
> 
> `Configure failed: Error: Bind 0x00124b0023adcfe3/1 msTemperatureMeasurement from '0x00124b0022813510/1' failed (AREQ - ZDO - bindRsp after 10000ms)`
> 
> I have another SNZB-02 in my network (added months ago) and it works fine.

Same here, bought 2 and both are not working, even with the screw driver fix.

**Comment by phobosdigg** [comment]
One of the sensors was connected using the maxbause method, but the other two sensors are not connected.

**Comment by Koenkk** [comment]
@takthetank well that is interesting since nothing has been changed in the device definition since then.

Any chance you could provide me the herdsman debug log when pairing on 1.19.1 (working) and on 1.21.2 (non-working)

See https://www.zigbee2mqtt.io/information/debug.html#zigbee-herdsman-debug-logging on how to enable the herdsman debug logging. Note that this is only logged to STDOUT and not to log files.

**Comment by Koenkk** [comment]
@takthetank I see temperature reports incoming from an unknown device:

```
2021-10-13T21:17:20.818766586Z   zigbee-herdsman:controller:log 'zcl' data is from unknown device with address '20312', skipping... +0ms
```

Do you have an other device which does temperature measurements in your network?

If not, maybe for some reason this device uses a different network address when pairing vs sending messages. Can you try:
- Force remove the SNZB-02
- Remove battery, and make sure the SNZB-02 is completely empty, press the reset button until it shows no signs of life anymore
- Pair the device again

**Comment by snippem** [comment]
@Koenkk bought 3 of these sensors as posted above the device pairs and then the following error apears.

Configure failed: Error: Bind 0x00124b0023205202/1 msTemperatureMeasurement from '0x00124b00214f3f95/1' failed (AREQ - ZDO - bindRsp after 10000ms)

Allready updated to the latest dev and did as you discribed:

Force remove the SNZB-02
Remove battery, and make sure the SNZB-02 is completely empty, press the reset button until it shows no signs of life anymore
Pair the device again

but no luck the device does pair but only shows null on al the values and shows the error when it pairs.
Have tried on a second pi4 and a cc2652 running hassos.
And the sensor does pair but also shows null on all the values and the same error. When i did the force remove then short circuit the sensor and pair again that works on that setup. It shows all of the values and it works as it should.
But this trick does not work on the setup that controlls my home. The only difference is that the zigbee stick is a cc2652RB and 124 devices on the network that the zigbee stick controlls.

**Comment by Koenkk** [comment]
@snippem 

> The only difference is that the zigbee stick is a cc2652RB and 124 devices on the network that the zigbee stick controlls.

Try pairing it very close to the coordinator, I think this device is a bit tricky over multiple hops.

**Comment by dsduarte** [comment]
> @snippem not sure how to fix it now, will check once I receive my SNZB-02.

I don't know if it helps, but I've deleted all my devices, installed the new edge version and after pairing using CC2531, SNZB-02 was reporting temp and humidity but the other parameteres like LQI and battery not.
I've deleted all again a reinstalled the edge version using sonoff zb bridge over TCP and all devices are reporting all of their parameters ok!

**Comment by salvq** [comment]
Just followed tipd from @maxbause and it just started working, values are being received finally


```
zigbee2mqtt    | Zigbee2MQTT:info  2021-10-17 13:43:56: MQTT publish: topic 'zigbee2mqtt/0x00124b00226b0cfb', payload '{"battery":23,"humidity":75.85,"linkquality":81,"temperature":28.2,"voltage":2800}'
zigbee2mqtt    | Zigbee2MQTT:info  2021-10-17 13:44:01: MQTT publish: topic 'zigbee2mqtt/0x00124b00226b0cfb', payload '{"battery":23,"humidity":75.85,"linkquality":81,"temperature":28.2,"voltage":2800}'
```

**Comment by pierrepinon** [comment]
@maxbause procedure works too, thx 🙏

**Comment by t0ny-peng** [comment]
Can confirm that the solution provided by maxbause works perfectly for my 3 sensors.

**Comment by snippem** [comment]
What are the network sizes with the maxbause way to add the sensor?
Because that way also works for me but only on a test setup with a few devices.
When I try to add them with the maxbause  way to a 124 device network it doesn't work anymore.
Then I get the same errors as dicribed in in the title of this topic.

**Comment by snippem** [comment]
> @maxbause your method works for me for my 8 sensors
> 
> For those having trouble, it's not clear from the picture but when your shorting the contacts what you are trying to short is the metal piece in the middle to the metal piece on the side. 

Please give more details about you setup.
Because I and others already stated.
The method works but how many routers and child devices are on the network.
As I stated before the method works but with my network with 124 routers and child devices it does not.

**Comment by dsduarte** [comment]
> @maxbause your method works for me for my 8 sensors
> 
> For those having trouble, it's not clear from the picture but when your shorting the contacts what you are trying to short is the metal piece in the middle to the metal piece on the side.

For me, what is not clear is between steps 3 and 5...
The method didn't work for me and I'm confused about the logic in delete the device from z2m and turn on the device... After deleting the device, z2m should allow it to regiser?

**Comment by grfernandes54** [comment]
@pniewiadowski your solution works for me.

**Comment by zold** [comment]
@pniewiadowski YES, this solution works for me.

**Comment by cperriot** [comment]
Hello, same probleme here,
some SNZB-02 not pairing, some pairing with no data retrieved.
i had to replace some batteries and to hold the sensor against the antenna.
having the sensor a few centimeters wasn't enough i press both to get everything ok...


coordinator texas 2652

**Comment by dsduarte** [comment]
I tried this "nearly touching" the antenna and worked for just one device... when I tried the second device it was reporting battery and LQI but N/A foi temp and humidity.
I don't know if there is an issue on my CC25231... I bought it already flashed and never changed anything...
I'm realy sad about this.. I realy like z2b and tried every combination using CC2531, Sonoff zigbee bridge over tcp, z2m 1.21.2-1, 1.19.1, z2m edge, z2m/z2m edge as HA Add-On, as docker container... all blind shots because i'm not a developer and could not troubleshooting the logs....  i think i'm done trying and pair all devices every new shot... I'll come back in a few weeks to see if there was any progress. Until that, if any one with knowledge to troubloshooting this, I'll be happy to perform some guided tests and provide the logs... besides the eagerness to solve this soon, I also will be happy to help the community.

**Comment by nebulorum** [comment]
> I have SONOFF SNZB-02 and the @maxbause solution didn't work for me. What I did is:
> 
> 1. Enable joining new devices in z2m
> 2. Press and hold paring button on Sonoff device
> 3. Wait until z2m shows error with configuration
> 4. Go to Device section and click yellow reconfigure button next to the Sonoff device
> 
> Then device starts reporting data finally.

I tried yellow reconfigure button and click on device and it did not work. Then popped the battery, clicked button on device, clicked yellow config, and kept clicking button on device until I say the successful configuration. Now I get a message every time I click the device button.

**Comment by tanmle** [comment]
I think the point here is just to keep hitting the buttons while re-configurating. :D

**Comment by TheSoldier91** [comment]
The Screwdriver hint works for me and 4 of my SNZB-02 devices.
Still struggling with SNZB-04, paired in z2m but no data is received, even if paired touching the ConBee II.

**Comment by hitterek** [comment]
Hello, how I was able to pair one SNZB-02 and one SNZB-03:
setup: rpi4, CC2531 usb dongle, zigbee2mqtt@1.22.0-dev (dev branch)

- stop z2m
- hold 5s button on cc2531
- start z2m
- reset device

first one was successfully paired (include temp). Second failed. So, I repeated all steps - stop z2, holt button, start, reset SNZB-03 and now everything works.

Error from snzb-03:
Zigbee2MQTT:error 2021-11-02 17:19:06: Failed to configure '0x00124b0023ad921f', attempt 2 (Error: Bind 0x00124b0023ad921f/1 genPowerCfg from '0x00124b001cd49e5e/1' failed (AREQ - ZDO - bindRsp after 10000ms)
    at Timeout._onTimeout (/home/pi/zigbee2mqtt-dev/zigbee2mqtt/node_modules/zigbee-herdsman/src/utils/waitress.ts:64:35)
    at listOnTimeout (node:internal/timers:557:17)
    at processTimers (node:internal/timers:500:7))

I will try to add more SNZB-02s tomorrow

**Comment by nightmare1111** [comment]
When pressing the recofigure yellow button , it is suggest to long hold the pairing button. Is this a 2 second hold ? as any longer removes the devuce and repairs it again with no temp reading.

**Comment by sjorge** [comment]
Would todays check-in and queue changes also maybe help for these devices to make pairing easier? @Koenkk

**Comment by Koenkk** [comment]
@sjorge I don't know if these support the check-in command since I have not received this device yet

**Comment by sjorge** [comment]
> @sjorge I don't know if these support the check-in command since I have not received this device yet

Those should show up as genPollCtrl cluster right? Then sadly now :(
<img width="361" alt="image" src="https://user-images.githubusercontent.com/379665/140718439-0ca9ddb6-8152-4b24-9bee-d40db61ea13c.png">

**Comment by Koenkk** [comment]
@sjorge doesn't support it indeed

**Comment by grfernandes54** [comment]
Hello, I've replaced my CC2531 coordinator. I've purchased a SMARTLIGHT CC2652P Zigbee USB dongle. After I replace de coordinator my SNZB-02 started to work for a while but they stopped to update de values after few hours. For make my network more secure I've change the network keys, this process requires to reparing all devices. During the reparing process all my devices worked except the SNZB-02. I have with the new coordinator the same error that I had with the previour coordinator

Configure failed: Error: Bind 0x00124b0023a5090a/1 msTemperatureMeasurement from '0x00124b0024bfd5f0/1' failed (AREQ - ZDO - bindRsp after 10000ms)

**Comment by vikdb** [comment]
Same here with a sonoff temp sensor.
I have 2 but only one fails..

**Comment by enigmaspb** [comment]
Good this recipe is working ! 
I rebind two  SNZB-02 
1. force delete on zigbee2mqtt dashboard
2. Bind then press join button in dashboard and long press button in SNZB-02 util 3 flash and more fast clicking button on  SNZB-02 .
3. Wait error message and press reconfigure on dashboard and more fast clicking button on  SNZB-02

**Comment by lesaintsgeosmois** [comment]
i have the same error with 2 sensors SNZB-02

msTemperatureMeasurement from '0x00124b001cdd4539/1' failed (AREQ - ZDO - bindRsp after 10000ms)

c2531
Version Zigbee2MQTT
    1.22.0 commit: 0a1fccf 

Type de Coordinateur
    zStack12

Révision du Coordinateur
    20201127

**Comment by Shellfishgene** [comment]
>     1. force delete on zigbee2mqtt dashboard
>     2. Press join button in dashboard and long press button in SNZB-02 util 3 flash and more fast clicking button on  SNZB-02 .
>     3. Wait error message and press reconfigure on dashboard and more fast clicking button on  SNZB-02

This worked for me too now with 3 sensors. For anyone trying, you need to press the button really fast continuously, like 3 times per second or so, at least for me. When you do that after resetting you may not even need to reconfigure and it works right away.

**Comment by dsduarte** [comment]
just received my new sonoff zigbee 3.0 USB dongle... started a fresh new instalation and paired all my sonoff's sensors and all of them are working perfectly from first attempt to pairing. No tricks... Just "Permit Join" and puting the device on pairing mode.

**Comment by dsduarte** [comment]
...forgot to mention... I was using a CC2531 adapter on a QNAP NAS, runing Home Assistant OS on a VM and having not only SNZB-02 issues but all of other SNZB's sensors were reporting something wrong. Nono of those tricks reported on this topic solved for me.
With this new sonoff zigbee CC2652 on my setup, all devices are working well.

**Comment by hot22shot** [comment]
Hi, same issue here.

For now I have a conbee2 plugged in my unRAID server but I ordered some sonoff zigbee USB stick to act as zigbee routers.
I was able to pair 2 of my SNZB-02, one without any issue the other one I had to use the methods described in this thread.

**Comment by craggus2000** [comment]
I'm having the same issue. Got 2x SNZB-02. connected via a generic CC2531 while I wait for a CC2652P2 PoE coordinator to be available. Both were working prefectly, now one of them is paired but not reporting. Tried re-pairing using all the ideas above, but still same problem persists. On version 1.22.1-1 with home assistant running on a generic x86-64 intel PC without any other issues.

**Comment by kds69** [comment]
Same here, procedure worked for previously paired (no need to force delete though) and new to be paired (zzh as coordinator).
`1. force delete on zigbee2mqtt dashboard
2. Press join button in dashboard and long press button in SNZB-02 util 3 flash and more fast clicking button on  SNZB-02 .
3. Wait error message and press reconfigure on dashboard and more fast clicking button on  SNZB-02`
thanks

**Comment by craggus2000** [comment]
Just update, this morning I had another go at this. Although I'd done this before, stopping zigbee2mqtt, removing and replacing the USB dongle, restarting zigbee2mqtt and re-pairing sensor now works. All reporting as expected. Wierd....

**Comment by Radlagerkok** [comment]
On version 1.18.1 these sensors worked fine.  They work very poorly on version 1.22.1.  You constantly have to reconnect.  They constantly freeze.  I cry

**Comment by camarril** [comment]
I was setting up sizbee2mqtt for the fist time yesterday, using the latest stable version (1.22.1) and my SNZB-02 only sent battery and signal strength information once after pairing and then stopped working. No temperature or humidity. Today I tried the latest-dev image (published 11 hours ago) and now it works perfectly.

**Comment by sanyafifa** [comment]
> Today I tried the latest-dev image (published 11 hours ago) and now it works perfectly.

it didn't work for me on the dev assembly.
Did you perform any actions after connecting?

**Comment by lesaintsgeosmois** [comment]
finaly it 's work for me :

Force Delete
Press the sensor reset buttom until flash for 3 times and wait to receive a
message at zigbee2qtt saying that the device left the network.
Press the join buttom at zigbee2mqtt
Press the sensor reset buttom until flash for 3 times and wait to receive a
message at zigbee2qtt saying that the device was configured
Wait for the error message
Press the reconfigure buttom

**Comment by camarril** [comment]
> Did you perform any actions after connecting?

No, I was able to pair it without any changes. Just pressed the button for 5 seconds and it worked.

**Comment by majkers** [comment]
I think in case of problems, the very first step should be restart zigbee2mqtt for example extension when using Home Assistant. I tried every scenario when my sensor stopped reporting and after a restart and reparing, it started to work.

**Comment by hot22shot** [comment]
I don't if this helps but I also bought some cheap USB dongles from SONOFF and adding one router did helps to connect with one of my SNZB-02 that did not want to connect.

For now my coordinator is a conbee II USB stick, I think I'll also replace my coordinator with a flashed SONOFF USB dongle.

**Comment by jbramwell** [comment]
> I have SONOFF SNZB-02 and the @maxbause solution didn't work for me. What I did is:
> 
>     1. Enable joining new devices in z2m
> 
>     2. Press and hold paring button on Sonoff device
> 
>     3. Wait until z2m shows error with configuration
> 
>     4. Go to Device section and click yellow reconfigure button next to the Sonoff device
> 
> 
> Then device starts reporting data finally.

This worked for me... thanks for the tip!

**Comment by nicolaerosia** [comment]
> Good this recipe is working ! I rebind two SNZB-02
> 
> 1. force delete on zigbee2mqtt dashboard
> 2. Bind then press join button in dashboard and long press button in SNZB-02 util 3 flash and more fast clicking button on  SNZB-02 .
> 3. Wait error message and press reconfigure on dashboard and more fast clicking button on  SNZB-02

this worked for me, thanks!

**Comment by fabgrenier** [comment]
Hi guys, I have the same problem and I tried to remove the device from domoticz, restart zigbeetomqtt, remove the battery etc etc but nothing worked for me.
Finally I manually edited `/opt/zigbee2mqtt/data/configuration.yaml` file to remove the faulty devices from the devices section and ... it worked ! 

Hope it helps.

**Comment by rollyfx** [comment]
Hi everyone
fabgrenier took me on the road.
I had try every procedures from web interface for re-pairing without success.
But if i restart zigbee2mqtt just after deleting and before re-pairing it work !

**Comment by linker3000** [comment]
Hi All, 

I have this problem too. I'm running a Node-RED based setup, with Mosquitto on Debian Linux. To start learning about Zigbee, I bought the SNZB-02, a SNZB-03 motion sensor and Sonoff Zigbee 3.0 USB Dongle Plus.

After initial setup I first paired the 02 and got temperature and humidity readings as expected. Later I paired up the 03 and that worked too. About an hour later I noticed I'd had no more data from the 02. Since then I must have tried every trick mentioned here to remove, re-pair, edit configs, restart zigbee2mqtt (systemctl) between remove/re-pair, take out batteries etc. and only the motion sensor still works.

The logs for zigbee2mqtt always show:

Configure failed: Error: Bind 0x00124b0025139274/1 msTemperatureMeasurement from '0x00124b0024cae404/1' failed (AREQ - ZDO - bindRsp after 10000ms)

Also: I am using the Sonoff USB Dongle Plus as it came out of the box, do I need to change firmware?

**Comment by domneedham** [comment]
> Hi everyone
> 
> fabgrenier took me on the road.
> 
> I had try every procedures from web interface for re-pairing without success.
> 
> But if i restart zigbee2mqtt just after deleting and before re-pairing it work !

Can confirm the exact same for me. 

- pair device to z2m
- force delete device
- restart z2m
- pair again

Works perfectly after the above

**Comment by fabgrenier** [comment]
Hi guys, since my last post, I had the same problem again with a sensor which is a bit far from my network. When it stops sending data, it never sends again despite trying to force pairing.

I ran into a simpler solution, just restart mosquito then zigbee2mqtt and the faulty device come back as normal. I didn't need to force remove from Domoticz.

**Comment by ndevenish** [comment]
After many attempts with 1.23 (with and without wiping everything except initial configuration), I downgraded to 1.20, (without wiping the database) which had the same error, and then 1.19 (after wiping the database)- which worked first time. I had tried several other of the suggestions in this thread, including maxbaus' suggestion.

This is my very first self-setup Zigbee device, which I bought with a CC2531 (I have hue bridge so probably background communication on separate network) so it's connecting directly to the CC2531.

**Comment by ndevenish** [comment]
So, I did some diving into this, and ran a bisect ([log](https://gist.github.com/ndevenish/733fd9722f523874a333d053cf3caa50)). Each stage in the bisect ran the docker build, started from a clear configuration.yaml (no devices, `network: GENERATE`), and I held down the pairing button until the light flashed and the zigbee2mqtt log showed activity. Sometimes I needed to pair twice, because (only on commits where it was broken), it sometimes didn't load even recognise the battery/linkquality/voltage fields.

The last (untagged) commit where it works (repeatedly tested) is https://github.com/Koenkk/zigbee2mqtt/commit/27b36b316c5fe5d23abfd083bee4ffc1b43070b9. The bisect reports it's in the next three commits: https://github.com/Koenkk/zigbee2mqtt/compare/27b36b316c5fe5d23abfd083bee4ffc1b43070b9...9f12ef7370fd75362b890f1cf2429a8faf96f1a1. Obviously my suspicion falls on https://github.com/Koenkk/zigbee2mqtt/commit/9f12ef7370fd75362b890f1cf2429a8faf96f1a1, "Update zigbee-herdsman to 0.13.146", because it seems unlikely to be caused by TypeScript restructuring.

However... the change to zigbee-herdsman in the public tag for those versions is.... not large: https://github.com/Koenkk/zigbee-herdsman/compare/v0.13.145...v0.13.146. The node-version at setup changed, and `Device.meta` was changed from private to public. Unfortunately, this is where the trail has run cold for me, because I don't know any of the cascading implications from either of those changes that might explain this error.

In any case, this implies that 1.21.1 is the last release where this worked; and it seems to work well for me, and pairs properly.

**Comment by sanyafifa** [comment]
Thank you @ndevenish !
downgraded to 1.21.1. Cleared all devices (copied the database file) Paired my snzb-02 and it worked!

**Comment by jjromannet** [comment]
I hit this issue few days ago when tried to setup 3 of my SNZB-02 devices (all purchased as a single order from single supplier). I only had coordinator in my network which is USB stick CC2531.
I read all comments in both issues here but nothing helped - SNZB-02s were only reporting battery level at pairing time and never actually actively send any message to coordinator. I tried hitting reconfigure along with pressing button at various frequencies, force remove devices and add them back, remove battery etc. For this purpose i was using device A & B, device C was only paired once and I did not touch it. 

From reading some comments here I concluded that people have more luck if there is router in the network. My another suspicion was that newer firmware (20201026) on SNZB-02 that i have is somehow incompatible. I purchased 2ch tuya switch device that is also acting as router and I gave it another go today. 
Today i had a bit more luck (it may be coincidence or indeed adding router to the network helped). Yet device A is not even reporting battery level right now and appears broken in the frontend - my working assumption is that I drain its batter during my tries last week. Device B started to report temperature and battery level (no humidity) only after i successfully reconfigured and set maximum reporting value for temperature to 60 (required frequent button presses). 
Device C - that i never tried to reconfigure before (only paired) - Today i only pressed reconfigure with default values  T=> min: 5, max: 1800, H => min: 10, max 3600 - then it started to report both T&H 
<img width="1175" alt="image" src="https://user-images.githubusercontent.com/1159276/154148241-9f09aaef-870f-4861-bdb9-7c7878b1a17f.png">
<img width="1058" alt="image" src="https://user-images.githubusercontent.com/1159276/154148339-cdcdb42a-9a77-43ac-8af5-013d69538126.png">

Another issue (not sure if related) is that "Last Seen" is always set to N/A even thought 2 out of 3 devices actually sending data to coordinator. 

My set up it to run standalone (without HA) zigbee2mqtt as a docker container with following versions:
```
Starting Zigbee2MQTT version 1.23.0 (commit #6ea61ba)
Starting zigbee-herdsman (0.14.10)
Coordinator firmware version: '{"meta":{"maintrel":3,"majorrel":2,"minorrel":6,"product":0,"revision":20201127,"transportrev":2},"type":"zStack12"}'
```
SNZB-02 firmware versions:  20201026
Linux kernel: 5.13.19-4

Attaching extracts from DB for all three devices [db-extract.zip](https://github.com/Koenkk/zigbee2mqtt/files/8074908/db-extract.zip)


I need to park the project for next two weeks. My next steps:
1. Test battery and order replacement 
2. Test version 1.21.1 
3. Try to understand what is interview and somehow force two remaining devices to complete it

**Comment by ndevenish** [comment]
> Try to understand what is interview and somehow force two remaining devices to complete it

Assuming that it’s not an easily traceable software issue, what would be required to properly diagnose this on the hardware side - sniffing the general packets was mentioned earlier, is that what would be definitive? (Which would presumably require buying a second coordinator and custom flashing)

**Comment by linker3000** [comment]
> 

I've also downgraded to 1.21.1 and the SNZB-02 has been working solid for a week now.

**Comment by danieldid** [comment]
I'm facing exactly the same issue with 1.21.1 through 1.23.

The error thrown in the logs:
`Configure failed: Error: Bind 0x00124b00245ce94c/1 msTemperatureMeasurement from '0x00212effff07710a/1' failed (waiting for response TIMEOUT)`

I have two Sonoff SNZB-02 from the same delivery. One is working like a charm. The other one is always having trouble after a couple of days. Usually removing and pairing helped out so far. Unfortunately this doesn't work anymore.

My setup:
- zigbee2mqtt@1.23
- conbee@0x26660700

I'm running everything in docker on a raspberry pi.

**Comment by jjromannet** [comment]
I downgraded to 1.21.1 and all of my SNZB-02s work now (even without router in the network). At first when i only changed version of image in docker-compose to `koenkk/zigbee2mqtt:1.21.1` and restarted with forced build it still was not working. 
I deleted all files apart from docker-compose and executed:
```
docker compose down --remove-orphans
docker compose rm
docker compose up -d
```
After this repaired all devices without  any issue. 
![image](https://user-images.githubusercontent.com/1159276/155890365-fbda4014-e52a-45cb-966d-88385d0d2bdd.png)

One of the differences i see is that in version 1.21.1 default accuracy for all reading is set to 2 decimal places while in most recent version it was set to no precision point. I wonder whether precision set to 0 may cause SNZB-02 to misbehave. 

This would explain why devices were not working after image downgrade - without repairing (during pairing new configuration is send). I will test this hypothesis after i am happy with result of 1.21.1 testing.

**Comment by danieldid** [comment]
@jjromannet I can confirm this works with 1.21.1

![2022-02-28 at 11 39](https://user-images.githubusercontent.com/4113271/155969064-d51fdb51-f551-46c6-bbe1-437d76df32f1.png)

By the way your temps are pretty hot (no offense! :-)) I'd like to have it cool
![2022-02-28 at 11 45](https://user-images.githubusercontent.com/4113271/155969967-0d9d14e1-550a-45fb-ab4d-af9ca2108b10.png)

As with 1.23 the interview doesn't work and no temp and humidity is delivered
![2022-02-28 at 11 56](https://user-images.githubusercontent.com/4113271/155971601-4f9c1247-a028-489a-b621-ad56d07a7837.png)

The difference (didn't check the changelogs!) between 1.21.1 and 1.23 is that you can't set the precission of the sensors under `Settings (spezifisch)` [german]. 

This is under 1.23
![2022-02-28 at 11 59](https://user-images.githubusercontent.com/4113271/155972022-cfcecf54-de23-4010-930b-243c64627e75.png)

This is under 1.21.1
![2022-02-28 at 12 00](https://user-images.githubusercontent.com/4113271/155972122-013e706d-3ae7-4410-a384-172c1d05696b.png)

I'll have a look on this for a couple of days and will be reporting.

**Comment by danieldid** [comment]
Update after 2 days: Everything's running normal. 

So it seems that downgrading was the solution or in other words in 1.23 something changed. I guess your thesis is correct @jjromannet. Must've been something with the accuracy. At least this is the only liable difference I can see.

BTW: I'm fairly new to docker and `--remove-orphans` did only work when using `docker-compose up -d --remove-orphans`. Should have read the flag first :-/ did reset some things (my complete influxdb datas were dropped).

![2022-03-02 at 21 44](https://user-images.githubusercontent.com/4113271/156446629-024e034d-3cbf-4006-9efd-9a501f23215f.png)

**Comment by DarkPako84** [comment]
> I have SONOFF SNZB-02 and the @maxbause solution didn't work for me. What I did is:
> 
> 1. Enable joining new devices in z2m
> 2. Press and hold paring button on Sonoff device
> 3. Wait until z2m shows error with configuration
> 4. Go to Device section and click yellow reconfigure button next to the Sonoff device
> 
> Then device starts reporting data finally.

Finally i have find a great solution!
Thanks!

**Comment by danieldid** [comment]
Post update - with 1.21.1 there's still no problem. Things are running normal.

**Comment by jjromannet** [comment]
@danieldid what do you mean by this? Did you setup your SNZB-02 with 1.21.1 and later upgraded zigbee2mqtt to more recent version - with SNZB-02 devices already paired?

**Comment by danieldid** [comment]
@jjromannet I set up everything with 1.21.1 in a docker environment. Then updated the image to the latest (back then) version 1.23. The problems mentioned here started with that version. I downgraded the docker image to 1.21.1 and started it with docker-compose up -d —remove-orphans (like mentioned previously).

Since then everything works seamlessly. 

Can anyone confirm that with the latest version 1.24 this is working properly?

**Comment by linker3000** [comment]
No 1.24 does not work. I upgraded to 1.21.2 and the sensor still worked,
but it stopped after a server reboot. I wanted 1.21.2 for Sonoff mains
switch support and I had to revert to 1.21.1 then upgrade to 1.21.2 again.

**Comment by dsduarte** [comment]
I had this problem again...
I removed all my SNZB-02's and started Z2M with the key "avalability = false" on config.yaml and paired again and it was ok!
I removed all my SNZB-02's again and started Z2M with "avalability = true" and paired again. Same error.
Than, at last I removed all my SNZB-02's and started Z2M with "avalability = false" and paired again and it was ok!

I think we can declare "avalability = false" on devices.yaml only for the SNZB-02's.

**Comment by jjromannet** [comment]
I think i will join "give up on SNZB-02" camp. After playing with initial three devices (out of which I though i bricked two) I bought another 2 (so 5 devices in total here). Two new ones join the network without any issues with software version 1.21.1 (no luck with newer versions). After a week one of them stopped reporting any data - by pure guess i bought new CR2450 quality batteries - with their help all of devices (including ones I initially though were broken) started to work properly. Only info I was able to find about original batteries shipped with SNZB-02 device ("Newsun" brand here) are three opinion on popular online store each for 1 star out of 5 - complaining battery dies after a week. 

Hint1: If you have any issues - start with new batteries - buy branded ones. 

Unfortunately even with new batteries indicators are randomly dying. 

![image](https://user-images.githubusercontent.com/1159276/160484124-be071ced-cdb3-46d9-8eac-06c667f5b4cf.png)
![image](https://user-images.githubusercontent.com/1159276/160484362-4371aac1-c4bc-4675-a990-18ac9ce30acd.png)
![image](https://user-images.githubusercontent.com/1159276/160484763-7d918ebd-83eb-450e-b75a-8d5c2419bd62.png)

**Comment by matbech** [comment]
I'm seeing the same issue as the original reported.

Also reconfigure does not solve my problem. The same error is reported:
```
info  2022-03-30 08:37:14: Configuring '0x00124b002512211d'
error 2022-03-30 08:37:24: Configure failed: Error: Bind 0x00124b002512211d/1 msTemperatureMeasurement from '0x00124b0022812a65/1' failed (AREQ - ZDO - bindRsp after 10000ms)
info  2022-03-30 08:37:24: Successfully configured '0x00124b002512211d'
info  2022-03-30 08:37:24: MQTT publish: topic 'zigbee2mqtt/bridge/response/device/configure', payload '{"data":{"id":"0x00124b002512211d"},"status":"ok"}'
```

**Comment by Koenkk** [comment]
@matbech seem to be a duplicate of https://github.com/Koenkk/zigbee2mqtt/issues/10973#issuecomment-1073053188 , already fixed in the dev branch (make sure to repair the device after upgrading).  (https://www.zigbee2mqtt.io/advanced/more/switch-to-dev-branch.html)

**Comment by danieldid** [comment]
I can confirm what @Koenkk said - this is fixed at least in latest dev branch. After reconfiguring all of my devices this works like a charm. This shall be closed though.

@matbech and @jjromannet did you tried it with 1.26 (latest) or dev branch?

**Comment by linker3000** [comment]
I've been running 1.25.0 for about a week and this version seems to have
fixed disconnect issues with both my SNZB-02 temperature sensors and Sonoff
S26R2ZB smart plugs.

Nigel

**Comment by ndevenish** [comment]
I've updated to 1.25.0 and it seems to be working fine. I needed to re-pair, reconfiguration failed, but the re-pair worked first time and running smoothly since then.

This also fixed the issues I had (on 1.21.0) with an Ikea Tradfri button, so pleased all around.

**Comment by Koenkk** [comment]
Thanks for confirming all, time to close this.

**Comment by MarcoGT** [comment]
With 1.30.1 not working

**Comment by dsduarte** [comment]
> With 1.30.1 not working

I got 1.30.3 and my SNZB-02 devices were getting offline and stop reporting every day...
Yesterday I forced removed them and then added again e now they are working well.

**Comment by miekrr** [comment]
Same, not working on the latest docker image. Following instructions https://github.com/Koenkk/zigbee2mqtt.io/commit/fd2285ca15ec45d394502af866d9a977e9791cef fixes temporary.

Next server reboot usually at least one sensor is left out and doesn't report anything.

**Comment by mnn** [comment]
I am on 1.32.2 (I think current latest?) and even the "temporary fix" doesn't work for me. It takes long just to start pairing (5-10 minutes), it has to be almost touching the antenna otherwise there's not a chance of even getting this "far". Then usually paring fails. And even when it doesn't fail, it doesn't setup reporting, so I never hear from the device again... Tried it two times, it took total half an hour and without any result :weary:.
